### ISSUE TITLE
Onboarding: display a clear message/CTA when website creation process fails

### ISSUE THREAD

**Issue by Robertght** [Automattic/wp-calypso#96827]
### Quick summary

Coming from [here](https://github.com/Automattic/dotcom-feedback/issues/308) and pfYzsZ-45-p2 I'm opening this report separately.

### Steps to reproduce

N/A

@ellatrix if you remember more about this and what steps we can take to replicate, please edit this section.

### What you expected to happen

To see a button to refresh or a message that the website creation process failed.

### What actually happened

The screen remained stuck on loading with errors in the browser console:

![Image](https://github.com/user-attachments/assets/68cacd20-d021-4371-967e-0d35803f50e5)

### Impact

One

### Available workarounds?

No and the platform is unusable

**Comment by jartes** [comment]
📌 REPRODUCTION RESULTS
- Could Not Replicate

📌 FINDINGS/SCREENSHOTS/VIDEO
- After following the steps on pfYzsZ-45-p2 I was not able to get the error.

📌 ACTIONS
- Leaving as `In Triage` waiting for reproduction steps.
- Lowering priority from `High` to `Low`.

**Comment by tanjoymor** [comment]
While I'm not sure about steps to reproduce, I can confirm that I have randomly run into this during other testing processes.

@paulopmt1 would you say this is related to what we ran into during the Quake plugin walk through?

It's also similar to what I ran into and commented on here: https://github.com/Automattic/wp-calypso/issues/96513#issuecomment-2508344865

Not sure what next steps should be here.

**Comment by paulopmt1** [comment]
> would you say this is related to what we ran into during the Quake plugin walk through?
> Could Not Replicate

By looking at the image, both `https://doubleclick.net/` and `public-api.wordpress.com` were unreachable by the user, which indicates they had a network connection error during that process. Since these resources are from two different companies, this seem to be related to the user end.

After understanding the flow and a few attempts to reproduce the issue, I found that if the `/onboarding-customation` route fails for any reason, we end up in this infinite loading screen. [This is the code responsible for it](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/site-setup-flow.ts#L224-L234).

To fix it, we need to handle that `Promisse.all` failure event so we can either try it again, or show a retry button to the user so [the window.location.assign](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/site-setup-flow.ts#L234) will be called as expected.

**Comment by Robertght** [comment]
Thank you @paulopmt1 !

I can mark this as Triaged on our board.

Would this be something Quake can look into moving forward?

**Comment by paulopmt1** [comment]
Yeah, I'll grab this on our board. Thanks for following up, @Robertght!
